File: LeetCode/Medium/2285-maximum-total-importance-of-roads/2285-maximum-total-importance-of-roads.go
func maximumImportance(n int, roads [][]int) int64 {
    count := make([]int, n)
    order := make([]int, n+1)
    var ret int64 = 0
    idx := n
    val := n

    for _, v := range roads {
        count[v[0]]++
        count[v[1]]++
    }

    for _, v := range count { order[v]++ }

    for idx > 0 {
        for idx > 0 && order[idx] < 1 { idx-- }
        if idx == 0 { break }
        order[idx]--
        ret += int64(idx * val)
        val--
    }

    return ret
}

File: LeetCode/Medium/2285-maximum-total-importance-of-roads/2285-maximum-total-importance-of-roads.java
class Solution {
    public long maximumImportance(int n, int[][] roads) {
        int[] count = new int[n];
        int[] order = new int[n+1];
        long ret = 0;
        int idx = n;
        long val = n;
        
        for (int[] v : roads){
            ++count[v[0]];
            ++count[v[1]];
        }
        
        for (int v : count) order[v]++;
        
        
        while (idx > 0){
            while (idx > 0 && order[idx] < 1) idx--;
            if (idx == 0) break;
            order[idx]--;
            ret += idx * val--;
        }
        
        return ret;
    }
}